### PR TITLE
Language: Powershell

### PR DESCRIPTION
**Pull request by JustinGrote**
Adds Powershell support via PSScriptAnalyzer

1. Adds Powershell into the DockerFile image
1. Creates a pwshlint.ps1 file to make PSScriptAnalyzer behave more "bash friendly" for how this program works
1. Supports exit codes
1. Has regex for powershell file extensions
1. Test-Automation test cases added for good ps1, bad ps1, psm1, and psd1

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,143 +1,178 @@
...
+RUN wget -O- -nvq https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s "$GO_VERSION"
+
+##################
+# Install TFLint #
+##################
+RUN curl -Ls "$(curl -Ls https://api.github.com/repos/terraform-linters/tflint/releases/latest | grep -o -E "https://.+?_linux_amd64.zip")" -o tflint.zip && unzip tflint.zip && rm tflint.zip \
+    && mv "tflint" /usr/bin/
+
+#########################################
+# Install Powershell + PSScriptAnalyzer #
```
Comment (TylerLeonhardt) [inline Dockerfile:102]: Powershell -> PowerShell

File: Dockerfile [hunk 2]
```diff
@@ -1,143 +1,178 @@
...
+#########################################
+# Reference: https://docs.microsoft.com/en-us/powershell/scripting/install/installing-powershell-core-on-linux?view=powershell-7
+# Slightly modified to always retrieve latest stable Powershell version
```
Comment (TylerLeonhardt) [inline Dockerfile:105]: Powershell -> PowerShell

File: README.md [hunk 3]
```diff
@@ -33,6 +33,7 @@ Developers on **GitHub** can call the **GitHub Action** to lint their code base
 | **TypeScript**   | [eslint](https://eslint.org/) [standard js](https://standardjs.com/)     |
 | **XML**          | [LibXML](http://xmlsoft.org/)                                            |
 | **YAML**         | [YamlLint](https://github.com/adrienverge/yamllint)                      |
+| **Powershell**   | [PSScriptAnalyzer](https://github.com/PowerShell/Psscriptanalyzer)       |
```
Comment (TylerLeonhardt) [inline README.md:36]: Powershell -> PowerShell

File: lib/linter.sh [hunk 4]
```diff
@@ -48,20 +48,24 @@ GO_LINTER_RULES="$DEFAULT_RULES_LOCATION/$GO_FILE_NAME"             # Path to th
 # Terraform Vars
 TERRAFORM_FILE_NAME='.tflint.hcl'                                        # Name of the file
 TERRAFORM_LINTER_RULES="$DEFAULT_RULES_LOCATION/$TERRAFORM_FILE_NAME"    # Path to the Terraform lint rules
+# Powershell Vars
+POWERSHELL_FILE_NAME='.powershell-psscriptanalyzer.psd1'                   # Name of the file
+POWERSHELL_LINTER_RULES="$DEFAULT_RULES_LOCATION/$POWERSHELL_FILE_NAME"    # Path to the Powershell lint rules
```
Comment (TylerLeonhardt) [inline lib/linter.sh:53]: Powershell -> PowerShell

File: lib/linter.sh [hunk 5]
```diff
@@ -48,20 +48,24 @@ GO_LINTER_RULES="$DEFAULT_RULES_LOCATION/$GO_FILE_NAME"             # Path to th
 # Terraform Vars
 TERRAFORM_FILE_NAME='.tflint.hcl'                                        # Name of the file
 TERRAFORM_LINTER_RULES="$DEFAULT_RULES_LOCATION/$TERRAFORM_FILE_NAME"    # Path to the Terraform lint rules
+# Powershell Vars
```
Comment (TylerLeonhardt) [inline lib/linter.sh:51]: Powershell -> PowerShell

File: lib/linter.sh [hunk 6]
```diff
@@ -1383,6 +1413,18 @@ BuildFileList()
       # Set the READ_ONLY_CHANGE_FLAG since this could be exec #
       ##########################################################
       READ_ONLY_CHANGE_FLAG=1
+    ###########################
+    # Get the Powershell files #
```
Comment (TylerLeonhardt) [inline lib/linter.sh:1417]: Powershell -> PowerShell

**Comment by admiralAwkbar** [comment]
Pulling this in as a branch for full ci and testing

**Comment by admiralAwkbar** [comment]
This has been pulled into #238

**Pull request closed without merging**